src/SWP/Component/Revision/Repository/RevisionRepositoryInterface.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Superdesk Web Publisher Templates System.
+ *
+ * Copyright 2015 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Component\Revision\Repository;
+
+use Doctrine\ORM\QueryBuilder;
+
+interface RevisionRepositoryInterface
+{
+    /**
+     * @return QueryBuilder
+     */
+    public function getPublishedRevision(): QueryBuilder;
```

Comment:
if we return something else than `QueryBuilder` from repository then we close way for extending it. On child class we would need duplicate code to do some modifications, this way we alway have very flexible query builder and if we don't want to work with `QueryBuilder` and `Query` in service or controller then we should create `Provider` for delivering result of repository.